# AI assignment

 ## 1.Write a program to find prime number with in a range using function.

In [1]:
# Function to check and return prime number
def isPrime(n):
    c=0
    for i in range(1,n+1):
        if n%i ==0:
            c+=1
    if c==2:
        return True
    else:
        return False

# Function to calculate prime number within given range
def primeCalculator(a,b):
    primeList = []
    for i in range(a,b+1):
        if isPrime(i):
            primeList.append(i)
    return primeList

# Asking for input
n1,n2 = map(int,input("Enter the range.").split(","))
if n1>n2:
    print(primeCalculator(n2,n1))
else:
    print(primeCalculator(n1,n2))

Enter the range. 10,20


[11, 13, 17, 19]


#### Enter the range. 10,20

## 2.Write a program to check palindrome.

In [2]:
# Function to check palindrome and return corresponding bool value 
def isPalindrome(n):
    rev = n[::-1]
    if rev == n:
        return True
    else:
        return False

# Main program
n = input("Enter the number/string to be checked.")
if isPalindrome(n):
    print("Palindrome")
else:
    print("Not palindrome")

Enter the number/string to be checked. 123454321


Palindrome


#### Enter the number/string to be checked. 123454321

## 3.Write a program to generate the Fibonacci sequence of a given number.

In [5]:
# Fibonacci number generator function
def fibonacci(n):
    ft = 0
    st = 1
    numberList = []
    for _ in range (n):
        numberList.append(ft)
        nt = ft + st
        ft = st
        st = nt
    return numberList

# Main program
number = int(input("Enter the number of terms."))
print(f"The fibonacci sequence is {fibonacci(number)}")

Enter the number of terms. 5


The fibonacci sequence is [0, 1, 1, 2, 3]


#### Enter the number of terms. 5

## 4.Create a number guessing game. Where you have to generate a random number in between 1 to 100 and user have to prompt a number for guess. if its high or low number let the user know and guide them through the process to guess correct number.

In [6]:
import numpy as np

mainLoop = True
while mainLoop:
    unknownNumber = np.random.randint(1,101)
    guessedNumber = int(input("Guess a number."))
    gameLoop = True
    choiceLoop = True
    while gameLoop:
        if guessedNumber < unknownNumber:
            print("Guess higher.")
            guessedNumber = int(input())
        elif guessedNumber > unknownNumber:
            print("Guess lower")
            guessedNumber = int(input())
        else:
            print("Correct guess")
            gameLoop = False
    print("Wana play again?")
    choice = input("y/n")
    while choiceLoop:
        if choice == 'y':
            choiceLoop = False
        elif choice == 'n':
            choiceLoop = False
            mainLoop = False
        else:
            print("Enter valid key")
            choice = input()

Guess a number. 45


Guess lower


 40


Guess lower


 35


Guess lower


 30


Guess lower


 25


Guess lower


 20


Guess lower


 15


Guess lower


 10


Guess higher.


 11


Guess higher.


 12


Guess higher.


 13


Guess higher.


 14


Correct guess
Wana play again?


y/n n


#### Guess a number. 45 40 35 30 25 15 10 11 12 13 14